analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd

```diff
@@ -81,21 +81,52 @@ lgat_specimens <- metadata %>%
   pull(Kids_First_Biospecimen_ID)
 ```
 
-# Read in snv consensus mutation data, filtering out LGAT
+# Read in snv consensus and hotspots mutation data, filtering out LGAT
+ - We will use the `pbta-snv-consensus-mutation.maf.tsv.gz` from `snv-callers` module which gathers calls that are present in all 3 callers (strelka2,mutect2 and lancet)
+ - In addition, we will also use `pbta-snv-scavenged-hotspots.maf.tsv.gz` from `hotspot-detection` module to gather calls that overlap MSKCC hotspots found in any caller (except if only vardict calls the site as variant, we remove these calls since we have a lot of calls unique to vardict which we consider as false positive as discussed [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/snv-callers#snv-caller-comparison-analysis))
+
 ```{r}
-snv_df <-
-  data.table::fread(file.path(root_dir,
-                              "data",
-                              "pbta-snv-consensus-mutation.maf.tsv.gz")) %>%
+
+snv_consensus_maf <- data.table::fread(
+  file.path(root_dir, "data" , "pbta-snv-consensus-mutation.maf.tsv.gz"),
+                                   select = c("Chromosome",
+                                              "Start_Position",
+                                              "End_Position",
+                                              "Strand",
+                                              "Variant_Classification",
+                                              "Tumor_Sample_Barcode",
+                                              "Hugo_Symbol",
+                                              "HGVSp_Short",
+                                              "Exon_Number"),
+                                   data.table = FALSE) 
+## Read in snv hotspot mutation data
+snv_hotspot_maf <- data.table::fread(
+  file.path(root_dir, "analyses" , "hotspots-detection" , "results" , "pbta-snv-scavenged-hotspots.maf.tsv.gz"),
+                                   select = c("Chromosome",
+                                              "Start_Position",
+                                              "End_Position",
+                                              "Strand",
+                                              "Variant_Classification",
+                                              "Tumor_Sample_Barcode",
+                                              "Hugo_Symbol",
+                                              "HGVSp_Short",
+                                              "Exon_Number"),
+                                   data.table = FALSE) %>%
+  select(colnames(snv_consensus_maf))
```

Comment:
Could you make the columns you're selecting a vector that you pass to `fread()` both times? Just thinking about if you were to want to add or drop a column you'd have to change it in two places.

Reply:
Ah yes, I'll add that! I did that in LGAT module  but not here , funny how my mind works :D